src/jacobians.jl

```diff
@@ -189,7 +190,11 @@ function finite_difference_jacobian(
                 _x1 = reshape(_vecx1,size(x))
                 vecfx1 = _vec(f(_x1))
                 dx = (vecfx1-vecfx)/epsilon
-                J = J + _make_Ji(J, eltype(x), dx, color_i, nrows, ncols)
+                if jac_prototype isa Nothing
+                    J = hcat(J, dx)
```

Comment:
doing a mapreduce would likely perform the `hcat` in a smarter way IIRC. This won't allocate as much as before, but it will still allocate quite a bit.